File: app/src/main/java/se/gustavkarlsson/aurora_notifier/android/background/providers/AuroraFactorsProvider.java
package se.gustavkarlsson.aurora_notifier.android.background.providers;

import android.location.Location;

import se.gustavkarlsson.aurora_notifier.android.models.AuroraFactors;

public interface AuroraFactorsProvider {
	AuroraFactors getAuroraFactors(Location location, long timeoutMillis);
}

File: app/src/main/kotlin/se/gustavkarlsson/skylight/android/services/providers/AuroraFactorsProvider.kt
package se.gustavkarlsson.skylight.android.services.providers

import io.reactivex.Single
import org.threeten.bp.Instant
import se.gustavkarlsson.skylight.android.entities.AuroraFactors
import se.gustavkarlsson.skylight.android.entities.Location

interface AuroraFactorsProvider {
	fun get(time: Single<Instant>, location: Single<Location>): Single<AuroraFactors>
}
